velox/functions/lib/LikeFunctionUtils.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/lib/LikeFunctionUtils.h"
+#include "velox/vector/TypeAliases.h"
+
+namespace facebook::velox::functions {
+namespace likeutils {
+
+bool isWildcardInPattern(const StringView& pattern) {
+  vector_size_t patternLength = pattern.size();
+  auto patternStr = pattern.data();
+  for (vector_size_t i = 0; i < patternLength; i++) {
+    if ((patternStr[i] == '%') || (patternStr[i] == '_')) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+bool matchExactPattern(const StringView& input, const StringView& pattern) {
+  vector_size_t inputLength = input.size();
+  vector_size_t patternLength = pattern.size();
+
+  if (inputLength != patternLength) {
+    return false;
+  }
+  return (std::memcmp(input.data(), pattern.data(), inputLength) == 0);
+}
+
+vector_size_t isPrefixPattern(const StringView& pattern) {
+  vector_size_t patternLength = pattern.size();
+  vector_size_t i = patternLength - 1;
+  auto patternStr = pattern.data();
+
+  for (; i >= 0; i--) {
+    if (patternStr[i] != '%') {
+      break;
+    }
+  }
+  if (i == patternLength - 1) {
+    return 0;
+  }
+  auto prefixLength = i + 1;
+
+  for (; i >= 0; i--) {
+    if ((patternStr[i] == '%') || (patternStr[i] == '_')) {
+      return 0;
+    }
+  }
+  return prefixLength;
+}
+
+bool matchPrefixPattern(
+    const StringView& input,
+    const StringView& pattern,
+    vector_size_t length) {
+  return (
+      std::memcmp(
+          input.getString().substr(0, length).c_str(),
```

Comment:
getString() returns an std::string. To do that it allocates new memory and copies the string into it. This data copy is not necessary.

`std::memcmp(input.data(), pattern.data(), length);`

Also, we need to check that input.length() >= length.